core/modules/image/image.field.inc

```diff
@@ -759,33 +822,153 @@ function image_field_formatter_view($entity_type, $entity, $field, $instance, $l
+      if ($is_svg) {
+        $image_svg_dimensions = $display['settings']['image_svg_dimensions'];
 
-    if (!empty($display['settings']['image_load'])) {
-      // Although available in Chromium, the 'auto' value is not mentioned in
-      // the specification. Since it may be subject to change, we recommend not
-      // to use it until it gets officially included.
-      if ($display['settings']['image_load'] != 'auto') {
-        $item['attributes']['loading'] = $display['settings']['image_load'];
+        foreach ($image_svg_dimensions as $attribute => $attribute_value) {
+          if (!empty($attribute)) {
+            $item[$attribute] = $attribute_value;
+          }
+        }
       }
     }
 
-    if (isset($link_file)) {
-      $uri = array(
-        'path' => file_create_url($item['uri']),
-        'options' => array(),
+    if (!$is_svg || $display['settings']['image_svg_display']) {
+      $element[$delta] = array(
+        '#theme' => 'image_formatter',
+        '#item' => $item,
+        '#image_style' => $display['settings']['image_style'],
+        '#path' => isset($uri) ? $uri : '',
       );
     }
-    $element[$delta] = array(
-      '#theme' => 'image_formatter',
-      '#item' => $item,
-      '#image_style' => $display['settings']['image_style'],
-      '#path' => isset($uri) ? $uri : '',
-    );
+    else {
+      $svg_path = file_create_url($item['uri']);
+      $svg = file_get_contents($svg_path);
+      $svg = preg_replace(['/<\?xml.*\?>/i', '/<!DOCTYPE((.|\n|\r)*?)">/i'], '', $svg);
+      $svg_clean = filter_xss($svg, image_field_allowed_svg_tags());
+      $svg_clean = trim($svg_clean);
+
+      if (isset($uri)) {
+        $element[$delta] = array(
+          '#type' => 'link',
+          '#href' => $uri['path'],
+          '#title' => $svg_clean,
+          '#options' => array(
+            'html' => TRUE,
+          ),
+        );
+      }
+      else {
+        $element[$delta] = array(
+          '#type' => 'markup',
+          '#markup' => $svg_clean,
+        );
+      }
+    }
   }
 
   return $element;
 }
+
+/**
+ * Provides a list of allowed SVG tags.
+ */
+function image_field_allowed_svg_tags() {
+  $tagsUc = array(
+    'a',
+    'animate',
+    'animateMotion',
+    'animateTransform',
+    'audio',
+    'canvas',
+    'circle',
+    'clipPath',
+    'defs',
+    'desc',
+    'discard',
+    'ellipse',
+    'feBlend',
+    'feColorMatrix',
+    'feComponentTransfer',
+    'feComposite',
+    'feConvolveMatrix',
+    'feDiffuseLighting',
+    'feDisplacementMap',
+    'feDistantLight',
+    'feDropShadow',
+    'feFlood',
+    'feFuncA',
+    'feFuncB',
+    'feFuncG',
+    'feFuncR',
+    'feGaussianBlur',
+    'feImage',
+    'feMerge',
+    'feMergeNode',
+    'feMorphology',
+    'feOffset',
+    'fePointLight',
+    'feSpecularLighting',
+    'feSpotLight',
+    'feTile',
+    'feTurbulence',
+    'filter',
+    'foreignObject',
+    'g',
+    'iframe',
```

Comment:
`foreignObject` is another candidate for removal here. The [default example on developer.mozilla.org](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/foreignObject) looks harmless, but there's more things one can do with it.

Reply:
The list is from here: https://www.w3.org/TR/SVG/eltindex.html

I dont know how we can reach consensus, but if you all dont mind I'm going to leave this list as is until there is agreement as to which should be in here and which should not, rather than pecking them out one by one (and probably putting them back in again). It'll only be a simple delete commit and push, so no rush.

Reply:
> I'm going to leave this list as is until there is agreement as to which should be in here and which should not...

Agreed. :+1: Let's discuss this properly.